### ISSUE TITLE
Flickering in menu/hanging after starting a game

### ISSUE THREAD

**Issue by sharpobject** [josefnpat/vapor#116]
These issues were observed using the .app from love2d.org, launching Vapor from Finder with right click -> open with. I prefer to launch from the terminal, and a bunch of much more serious issues appear if I do, but they seem to be bugs in LÖVE, so I'm not reporting them here.
- The game image flickers rapidly when selecting games.
- Vapor always hangs while playing a game. Activity Monitor's quit/kill will not kill it, but Activity Monitor's force quit/kill -9 will.

**Comment by josefnpat** [comment]
Thank you for this issue request. **Next time when you submit tickets, please submit one ticket per issue.**

> bunch of much more serious issues appear if I do

Please post this information. I can tell you if they are coming from LÖVE or Vapor.

> The game image flickers rapidly when selecting games.

I believe this may be a duplicate of #87? Please tell me if it is or is not.

> Vapor always hangs while playing a game.

I assume you're running OSX. I think this can be fixed by forking the instance of the game on *nix systems when the `os.execute` is called.

```
local fstr = [["%s" "%s/%s"]]
```

change to

```
local fstr = [["%s" "%s/%s &"]]
```

https://github.com/josefnpat/vapor/blob/master/src/core/vapor.lua#L20

Here's the patch:

```
diff --git a/src/core/vapor.lua b/src/core/vapor.lua
index c26403a..bb889a9 100644
--- a/src/core/vapor.lua
+++ b/src/core/vapor.lua
@@ -17,7 +17,7 @@ function vapor.execgame(binarypath, gamepath)
     execstr = fstr:format(binarypath, gamepath)
   else
     -- OS X, Linux
-    local fstr = [["%s" "%s/%s"]]
+    local fstr = [["%s" "%s/%s &"]]
     execstr = fstr:format(binarypath, love.filesystem.getSaveDirectory(), gamepath)
   end
   print(gamepath.." starting.")
```

I will test this out at home once I get the chance. If you feel so inclined, you can tell me if this change fixes your problem.

**Comment by sharpobject** [comment]
I'll split issues up better in the future :)

> Please post this information. I can tell you if they are coming from LÖVE or Vapor.

I am pretty sure they are coming from LÖVE. 2/3 of them are tested as occurring without Vapor. The mouse issue was only observed in Vapor, as the other games I tested did not use the mouse. See the report here: https://bitbucket.org/rude/love/issue/798/super-broken-when-launching-080-from

> I believe this may be a duplicate of #87? Please tell me if it is or is not.

That is correct; it is a duplicate of #87.

Regarding the os.execute fix, I expect it to work, but I am not completely certain because I'm really only familiar with luaposix. I'm not able to test until this evening.

**Comment by josefnpat** [comment]
Ok, from now on this ticket will not be a part of #87 and will only go on to deal with vapor being blocked by games that are being run.

Fix incoming.

**Comment by josefnpat** [comment]
@sharpobject please confirm this fix works on OS X if you can.

**Comment by sharpobject** [comment]
This works now :)
